src/db.ts

```diff
@@ -0,0 +1,138 @@
+import { Properties } from '@posthog/plugin-scaffold'
+import { ClickHouse } from 'clickhouse'
+import { Producer } from 'kafkajs'
+import { DateTime } from 'luxon'
+import { Pool, PoolConfig } from 'pg'
+import { KAFKA_PERSON, KAFKA_PERSON_UNIQUE_ID } from './ingestion/topics'
+import { unparsePersonPartial } from './ingestion/utils'
+import { Person, PersonDistinctId, RawPerson } from './types'
+import { castTimestampOrNow } from './utils'
+
+export class DB extends Pool {
```

Comment:
I think this is putting too many responsibilities together. To follow the same pattern we have with each other service we connect to, I'd still keep a `db` (or `postgres`) pure DB connection object on the `server`.

The code in this file could be split into a new `orm` (or `db`) class and then just take `postgres` as another input.

Reply:
I dunno, I wouldn't want to decouple them actually… More complexity in `createServer` and we want to use these methods every single time, so I'm in favor of keeping this integrated.

Reply:
I'd still push for separating `postgres` out of this class. This `DB` is now a meta class that uses all different database engines for its work, so extending postgres's `Pool` doesn't really make sense to me, even if that's the database this class mostly uses.

It's hard to explain why the first one in these is without `.postgres`:

```
this.query
this.clickhouse.query
this.kafkaProducer.send
```

I mean, extending classes is usually a `Object > Mammal > Dog` type of thing. Here it's doing `Object > Mammal > Dog & Reptile` 😅

Reply:
Very well then, separated

Reply:
Tests were failing as a lot of places still called `server.db.query`. I swapped them to `server.db.postgresQuery`. 

We don't have just one `db` unfortunately, so keeping `server.db.query` like it was now adds cognitive steps. The alternatives were `server.db.postgresQuery` and `server.postgres.query`... and of those I found `server.db.postgresQuery` a better fit, since we have so many other methods now on `db.*`, and it's easy to just pass around one object... Hope that makes sense 😅